demos/multi_material/compositional_buoyancy/van_keken_1997.py

```diff
@@ -0,0 +1,276 @@
+T = Function(Q, name="Temperature")  # Firedrake function for temperature
+psi = Function(K, name="Level set")  # Firedrake function for level set
+# -
+
+# We now provide initial conditions for the level set field. To this end, we use the
+# shapely library to represent the initial location of the material interface and
+# derive the signed-distance function. Finally, we apply the transformation to obtain a
+# smooth step function profile.
+
+# +
+import numpy as np
+import shapely as sl
+
+
+def cosine_curve(x, amplitude, wavelength, vertical_shift):
+    """Cosine curve equation with an amplitude and a vertical shift"""
+    return amplitude * np.cos(2 * np.pi / wavelength * x) + vertical_shift
+
+
+interface_deflection = 0.02  # Amplitude of the cosine function marking the interface
+interface_wavelength = 2 * lx  # Wavelength of the cosine function
+material_interface_y = 0.2  # Vertical shift of the interface along the y axis
+# Group parameters defining the cosine profile
+isd_params = (interface_deflection, interface_wavelength, material_interface_y)
+
+# Shapely LineString representation of the material interface
+interface_x = np.linspace(0, lx, 1000)
+interface_y = cosine_curve(interface_x, *isd_params)
+line_string = sl.LineString([*np.column_stack((interface_x, interface_y))])
+sl.prepare(line_string)
+
+# Extract node coordinates
+node_coords_x, node_coords_y = node_coordinates(psi)
+# Determine to which material nodes belong and calculate distance to interface
+node_relation_to_curve = [
+    (
+        node_coord_y > cosine_curve(node_coord_x, *isd_params),
+        line_string.distance(sl.Point(node_coord_x, node_coord_y)),
+    )
+    for node_coord_x, node_coord_y in zip(node_coords_x, node_coords_y)
+]
+
+# Define the signed-distance function and overwrite its value array
+signed_dist_to_interface = Function(K)
+signed_dist_to_interface.dat.data[:] = [
+    dist if is_above else -dist for is_above, dist in node_relation_to_curve
+]
+
+# Define thickness of the hyperbolic tangent profile
+min_mesh_edge_length = min(lx / nx, ly / ny)
+epsilon = Constant(min_mesh_edge_length / 4)
+
+# Initialise level set as a smooth step function
+psi.interpolate((1 + tanh(signed_dist_to_interface / 2 / epsilon)) / 2)
+# -
+
+# We next define materials present in the simulation. Here, the problem is
+# non-dimensionalised and can be described by the product of the expressions for the
+# Rayleigh and buoyancy numbers, RaB, which is also referred to as compositional
+# Rayleigh number. Therefore, we provide a value for thermal and compositional Rayleigh
+# numbers to define our approximation.
+
+# +
+buoyant_material = Material(RaB=-1)  # Vertical direction is flipped in the benchmark
+dense_material = Material(RaB=0)
+materials = [buoyant_material, dense_material]
+
+Ra = 0  # Thermal Rayleigh number
+
+RaB = field_interface(
+    [psi], [material.RaB for material in materials], method="arithmetic"
```

Comment:
Worth adding a comment here about the "arithmetic" averaging approach used at the interface (I assume that's what this is?) alongside the equation? Do you perhaps want to quickly note the other options available (geometric, harmonic) and those equations too?